File: src/main/java/com/bdev/hogwarts_api/data/dto/student/StudentPayment.java
package com.bdev.hogwarts_api.data.dto.student;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Builder;
import lombok.Getter;
import lombok.NonNull;

@Builder
@Getter
public class StudentPayment {
    private Long id;
    @NonNull private Long studentId;
    @NonNull private Long amount;
    @NonNull private Long time;

    @JsonCreator
    public static StudentPayment create(
            @JsonProperty("id") Long id,
            @JsonProperty("studentId") Long studentId,
            @JsonProperty("amount") Long amount,
            @JsonProperty("time") Long time
    ) {
        return StudentPayment
                .builder()
                .id(id)
                .studentId(studentId)
                .amount(amount)
                .time(time)
                .build();
    }

}

File: src/main/java/com/bdev/hogwarts_api/data/dto/student/StudentPayment.kt
package com.bdev.hogwarts_api.data.dto.student

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

class StudentPayment @JsonCreator constructor(
        @JsonProperty("id") val id: Long?,
        @JsonProperty("studentId") val studentId: Long,
        @JsonProperty("teacherId") val teacherId: Long,
        @JsonProperty("amount") val amount: Long,
        @JsonProperty("time") val time: Long,
        @JsonProperty("processed") val processed: Boolean
)
